File: Excercise1/task1.c
// gcc 4.7.2 +
// gcc -std=gnu99 -Wall -g -o helloworld_c helloworld_c.c -lpthread

#include <pthread.h>
#include <stdio.h>

int i = 0;

// Note the return type: void*
void* thread1_func(){
  for(int j=0; j<1000000;j++){
  		i++;
  	}
    return NULL;
}

void* thread2_func(){
   for(int j=0; j<1000000;j++){
  		i--;
	}
    return NULL;
}


int main(){
    pthread_t thread1;
    pthread_t thread2;
    pthread_create(&thread1, NULL, thread1_func, NULL);
    pthread_create(&thread2, NULL, thread2_func, NULL);
    // Arguments to a thread would be passed here ---------^
    
   if(!pthread_join(thread1, NULL))
   		printf("Thread1_join sucsess\n");
    if(!pthread_join(thread2, NULL))
    	printf("Thread2_join sucsess\n");
    printf("Hello from main!\n");
    printf("i = %d\n",i);
    printf("Hello from Big Jonny");
    return 0;
    
}

OUTPUT
	Success	time: 0 memory: 18632 signal:0
	Thread1_join sucsess
	Thread2_join sucsess
	Hello from main!
	i = 0

File: Excercise1/task1.py
# Python 3.3.3 and 2.7.6
# python helloworld_python.py

from threading import Thread

i = 0

def thread1_func():
	global i
	print("Hello from a thread1!")
	for j in range (0,1000000):
   		i +=1
   	print("i =+ %d" % i)

def thread2_func():
	global i
	print("Hello from a thread2!")
	for k in range (0,1000000):
		i -=1
	print("i -= %d" % i) 


def main():
    thread1 = Thread(target = thread1_func, args = (),)
    thread2 = Thread(target = thread2_func, args = (),)
    thread1.start()
    thread2.start()
    thread1.join()
    thread2.join()
    print("Hello from main!")
    print("Hello from testeditor2!")
    print("i = %d" % i) 
   
main()